core/math/a_star_grid_2d.cpp

```diff
@@ -734,6 +734,119 @@ TypedArray<Vector2i> AStarGrid2D::get_id_path(const Vector2i &p_from_id, const V
 	return path;
 }
 
+Vector2 AStarGrid2D::get_flow_direction(const Vector2i &p_id) const {
+	ERR_FAIL_COND_V_MSG(dirty, Vector2(), "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!is_in_boundsv(p_id), Vector2(), vformat("Can't get point's flow direction. Point %s out of bounds %s.", p_id, region));
+	return _get_point_unchecked(p_id)->flow_direction;
+}
+
+TypedArray<Vector2> AStarGrid2D::get_flow_directions(const Rect2i &p_region) const {
+	ERR_FAIL_COND_V_MSG(dirty, TypedArray<Vector2i>(), "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!region.encloses(p_region), TypedArray<Vector2i>(), vformat("Can't get point's flow direction. Region %s out of bounds %s.", p_region, region));
+
+	TypedArray<Vector2> result;
+	result.resize(p_region.size.x * p_region.size.y);
+
+	uint32_t index = 0U;
+	for (int y = p_region.get_position().y; y < p_region.get_end().y; y++) {
+		for (int x = p_region.get_position().x; x < p_region.get_end().x; x++) {
+			result[index++] = _get_point_unchecked(Vector2i(x, y))->flow_direction;
```

Comment:
`result[index++] = _get_point_unchecked(x, y)->flow_direction;` 
Would even be better to have the raw pointer somehow, to avoid all the boundchecks for every access, but not sure it's doable in the current state

Reply:
I said you that it's not possible with TypedArray

Reply:
The main point of this comment is to remove the superflous `Vector2i` constructor call. The other part is just a comment on the API, yo can just ignore it.

Reply:
I've simply removed that method - feel free to implement it yourself.